src/errors.rs

```diff
@@ -0,0 +1,180 @@
+//! Error definitions
+
+pub mod i2c {
+    /// I2C error
+    pub trait Error: core::fmt::Debug {
+        /// Convert error to a generic I2C error kind
+        ///
+        /// By using this method I2C errors freely defined by HAL implementations
```

Comment:
Nitpick: Comma missing.

```suggestion
        /// By using this method, I2C errors freely defined by HAL implementations
```

(Same thing applies to other `Error` definitions below.)